helm-charts/splunk-otel-collector/scripts/lookup-eks-fargate-receiver-node.sh

```diff
@@ -0,0 +1,48 @@
+#! /usr/bin/bash
+set -ex
+
+# If we are the first pod (cluster receiver), set the kubelet stats node filter to only follow labelled nodes.
+# This node label will be set by the second pod.
+if [[ "${K8S_POD_NAME}" == *-0 ]]; then
+  echo "will configure kubelet stats receiver to follow node ${FIRST_CR_REPLICA_NODE_NAME}, as well as use cluster receiver."
+  echo "export CR_KUBELET_STATS_NODE_FILTER='&& labels[\"splunk-otel-is-eks-fargate-cluster-receiver-node\"] == \"true\"'" >/splunk-messages/environ
+  cat /splunk-messages/environ
+
+  # copy config to meet container command args
+  cp /conf/relay.yaml /splunk-messages/config.yaml
```

Comment:
If we are copying the config anyway, maybe we can adjust the config same way in both place with yq (the discovery rule and receivers list) instead of using /splunk-messages/environ. I think it should make it simpler.

Reply:
Using the env vars seemed more accommodating of future changes to the underlying config, but I'm trying this out.

Reply:
Maybe more transparent but I think it complicates the logic. Would be nice to reduce the number of scripts and configmap mounts

Reply:
updated to use yq